bitwarden_license/test/Scim.IntegrationTest/Controllers/v2/UsersControllerTests.cs

```diff
@@ -0,0 +1,350 @@
+            Assert.Equal(10, responseModel.ItemsPerPage);
+            Assert.Equal(1, responseModel.TotalResults);
+            Assert.Equal(1, responseModel.StartIndex);
+
+            var user = responseModel.Resources.Single();
+            Assert.Equal(ScimApplicationFactory.TestOrganizationUserId3.ToString(), user.Id);
+            Assert.Equal("Test User 3", user.DisplayName);
+            Assert.Equal("UC", user.ExternalId);
+            Assert.Equal(new List<string> { ScimConstants.Scim2SchemaListResponse }, responseModel.Schemas);
+        }
+
+        [Fact]
+        public async Task Post_Success()
+        {
+            var organizationId = ScimApplicationFactory.TestOrganizationId1;
+            var model = new ScimUserRequestModel
+            {
+                DisplayName = "New User",
+                Emails = new List<BaseScimUserModel.EmailModel> { new BaseScimUserModel.EmailModel("user4@mail.com") },
+                ExternalId = null,
+                Schemas = null,
+                Active = true
+            };
+
+            var context = await _factory.UsersPostAsync(organizationId, model);
+
+            Assert.Equal(StatusCodes.Status201Created, context.Response.StatusCode);
+
+            var databaseContext = _factory.GetDatabaseContext();
+            Assert.Equal(4, databaseContext.OrganizationUsers.Count());
+        }
+
+        [Theory]
+        [InlineData(null)]
+        [InlineData("")]
+        [InlineData(" ")]
+        public async Task Post_InvalidDisplayName_BadRequest(string displayName)
+        {
+            var organizationId = ScimApplicationFactory.TestOrganizationId1;
+            var model = new ScimUserRequestModel
+            {
+                DisplayName = displayName,
+                ExternalId = null,
+                Schemas = null,
+                Active = true
+            };
+
+            var context = await _factory.UsersPostAsync(organizationId, model);
+
+            Assert.Equal(StatusCodes.Status400BadRequest, context.Response.StatusCode);
+        }
+
+        [Fact]
+        public async Task Post_ExistingEmail_Conflict()
+        {
+            var organizationId = ScimApplicationFactory.TestOrganizationId1;
+            var model = new ScimUserRequestModel
+            {
+                DisplayName = "New User",
+                Emails = new List<BaseScimUserModel.EmailModel> { new BaseScimUserModel.EmailModel("user1@mail.com") },
+                ExternalId = "UA",
+                Schemas = null,
+                Active = true
+            };
+
+            var context = await _factory.UsersPostAsync(organizationId, model);
+
+            Assert.Equal(StatusCodes.Status409Conflict, context.Response.StatusCode);
+
+            var databaseContext = _factory.GetDatabaseContext();
+            Assert.Equal(3, databaseContext.OrganizationUsers.Count());
+        }
+
+        [Fact]
+        public async Task Put_RevokeUser_Success()
```

Comment:
I opted to have them separate as separate tests because one is revoking access to an active user and the other one is restoring access to a user that is already disabled.